### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by Elhambasir**
### What is done in this PR?

- [x] Created a table named vets with the following columns:
      id: integer (set it as autoincremented PRIMARY KEY)
      name: string
      age: integer
      date_of_graduation: date

- [x] Created a "join table" called specializations
- [x] Created a "join table" called visits
- [x] Insert the following data for vets:
        Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
        Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
        Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
        Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.

- [x] Insert the following data for specialties:
        Vet William Tatcher is specialized in Pokemon.
        Vet Stephanie Mendez is specialized in Digimon and Pokemon.
        Vet Jack Harkness is specialized in Digimon.

- [x] Insert the following data for visits:
      Agumon visited William Tatcher on May 24th, 2020.
      Agumon visited Stephanie Mendez on Jul 22th, 2020.
      Gabumon visited Jack Harkness on Feb 2nd, 2021.
      Pikachu visited Maisy Smith on Jan 5th, 2020.
      Pikachu visited Maisy Smith on Mar 8th, 2020.
      Pikachu visited Maisy Smith on May 14th, 2020.
      Devimon visited Stephanie Mendez on May 4th, 2021.
      Charmander visited Jack Harkness on Feb 24th, 2021.
      Plantmon visited Maisy Smith on Dec 21st, 2019.
      Plantmon visited William Tatcher on Aug 10th, 2020.
      Plantmon visited Maisy Smith on Apr 7th, 2021.
      Squirtle visited Stephanie Mendez on Sep 29th, 2019.
      Angemon visited Jack Harkness on Oct 3rd, 2020.
      Angemon visited Jack Harkness on Nov 4th, 2020.
      Boarmon visited Maisy Smith on Jan 24th, 2019.
      Boarmon visited Maisy Smith on May 15th, 2019.
      Boarmon visited Maisy Smith on Feb 27th, 2020.
      Boarmon visited Maisy Smith on Aug 3rd, 2020.
      Blossom visited Stephanie Mendez on May 24th, 2020.
      Blossom visited William Tatcher on Jan 11th, 2021.

###  Check the query results bellow: 

Who was the last animal seen by William Tatcher?
who-last-visit-william

How many different animals did Stephanie Mendez see?  
How many different animals did Stephanie Mendez see

List all vets and their specialties, including vets with no specialties.
List all vets and their specialties, including vets with no specialties

List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020

What animal has the most visits to vets?
What animal has the most visits to vets

Who was Maisy Smith's first visit?
Who was Maisy Smith's first visit

Details for most recent visit: animal information, vet information, and date of visit.
Details for most recent visit animal information vet information and date of visit

How many visits were with a vet that did not specialize in that animal's species?
opossit

What specialty should Maisy Smith consider getting? Look for the species she gets the most.
What specialty should Maisy Smith consider getting Look for the species she gets the most

### Thank you for reviewing :

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -151,3 +151,94 @@ owners ON
...
+     species_id FROM 
+     animals WHERE 
+     id = animal_id)) FROM 
+     visits WHERE 
+     vet_id = (select 
+     id from 
+     vets WHERE 
+     name = 'Maisy Smith') GROUP BY 
+     animal_id ORDER BY 
+     c DESC LIMIT 1 ;
```
Comment (amejid) [inline queries.sql:244]: - [x] You did a great job setting up the JOIN table for visits. :+1: I think some of the queries are not returning the results we are expecting. Let's update them to return the correct queries. ✍️ According to the data given below, six (6) Digimon visited Maisy Smith. Kindly update your query so it returns 6. 🤝️

File: queries.sql [hunk 2]
```diff
@@ -151,3 +151,94 @@ owners ON
...
+    vets ON 
+    visits.vet_id = vets.id ORDER BY 
+    visits.date_of_visit DESC LIMIT 1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+    SELECT COUNT(visits.date_of_visit) FROM 
+    visits JOIN 
+    vets ON 
+    visits.vet_id = vets.id WHERE 
+    vets.id NOT IN (SELECT specializations.vet_id FROM specializations);
```
Comment (amejid) [inline queries.sql:227]: - [x]  Kindly update this query to return twelve(12). I believe that is the correct result. Let's see the breakdown below. 🤝️
  - 9 animals visited Maisy Smith but she did not specialize in any of the species 
  - 2 animals (Agumon and Plantmon) are Digimon visited William Thatcher while he specializes in Pokemon
  - 1 animal (Charmander) is Pokemon that visited Jack Harkness while he specializes in Digimon

This makes the number of visits with a vet that did not specialize in that animal's species 12

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @Elhambasir, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Correct workflow :heavy_check_mark: 
- SQL organized in multiple files :heavy_check_mark: 
- JOIN table correctly set up :heavy_check_mark: 
- Most queries returning correct results :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Elhambasir) [reply queries.sql:244]: Ok

File: queries.sql (hunk 2 above)
Comment (Elhambasir) [reply queries.sql:227]: Ok

**Review by Zuhaib042 — APPROVED** [review]
Hi @Elhambasir 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Implemented suggested changes perfectly
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**